aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -432,4 +432,74 @@ Tensor _logcumsumexp_cuda(const Tensor& self, int64_t dim) {
   return _logcumsumexp_out_cuda(result, self, dim);
 }
 
+Tensor& _cumsum_out_cuda(Tensor& result, const Tensor& self, int64_t dim) {
+  result.resize_(self.sizes());
+  if (self.dim() == 0) {
+    result.fill_(self);
+    return result;
+  }
+  if (self.numel() == 0) {
+    result.zero_();
+    return result;
+  }
+  auto wrap_dim = maybe_wrap_dim(dim, self.dim());
+
+  TensorArg output_arg{result, "output", 1};
+  TensorArg input_arg{self, "input", 2};
+  checkAllSameGPU("cumsum", {output_arg, input_arg});
+
+  AT_DISPATCH_ALL_TYPES_AND(
+      at::ScalarType::Half, self.scalar_type(), "cumsum_cuda", [&]() {
```

Comment:
this assumes `self` and `result` have the same datatype, but I don't see it being checked anywhere. When users supply `out` argument it can be of any type.

Reply:
Makes sense. Will add a check.